tests/dragonfly/connection_test.py

```diff
@@ -0,0 +1,176 @@
+    conn = aioredis.Redis(connection_pool=pool)
+    pipe = conn.pipeline()
+    for key, val in messages.items():
+        pipe.set(key, val)
+    result = await pipe.execute()
+
+    print(f"got result from the pipeline of {result} with len = {len(result)}")
+    if len(result) != len(messages):
+        return False, f"number of results from pipe {len(result)} != expected {len(messages)}"
+    elif False in result:
+        return False, "expecting to successfully get all result good, but some failed"
+    else:
+        return True, "all command processed successfully"
+
+
+'''
+Run test in pipeline mode.
+This is mostly how this is done with python - its more like a transaction that
+the connections is running all commands in its context
+'''
+
+
+def test_pipeline_support(async_pool):
+    def generate(max):
+        for i in range(max):
+            yield f"key{i}", f"value={i}"
+
+    messages = {a: b for a, b in generate(5)}
+    loop = asyncio.new_event_loop()
+    success, message = loop.run_until_complete(
+        run_pipeline_mode(async_pool, messages))
+    assert success, message
+
+
+async def reader(channel: aioredis.client.PubSub, messages):
+
+    success = True
+    message_count = len(messages)
+    while message_count > 0:
+        try:
+            async with async_timeout.timeout(1):
+                message = await channel.get_message(ignore_subscribe_messages=True)
+                if message is not None:
+                    message_count = message_count - 1
+                    #print(f"(Reader) Message Received: {message}")
+                    if message["data"] not in messages:
+                        return False, f"got unexpected message from pubsub - {message['data']}"
+                await asyncio.sleep(0.01)
+        except asyncio.TimeoutError:
+            pass
+    return True, "success"
+
+
+async def run_pubsub(pool, messages, channel_name):
+    conn = aioredis.Redis(connection_pool=pool)
+    pubsub = conn.pubsub()
+    await pubsub.subscribe(channel_name)
+
+    future = asyncio.create_task(reader(pubsub, messages))
+    success = True
+
+    for message in messages:
+        res = await conn.publish(channel_name, message)
+        if not res:
+            success = False
+            break
+
+    await future
+    status, message = future.result()
+    if status and success:
+        return True, "successfully completed all"
+    else:
+        return False, f"subscriber result: {status}: {message},  publisher publish: success {success}"
+
+''' 
+Test the pipeline command
+Open connection to the subscriber and publish on the other end messages
+Make sure that we are able to send all of them and that we are getting the
+expected results on the subscriber side
+'''
+
+
+def test_pubsub_command(async_pool):
+    def generate(max):
+        for i in range(max):
+            yield f"message number {i}"
+
+    messages = [a for a in generate(5)]
+    loop = asyncio.new_event_loop()
```

Comment:
i have googled pytest and asyncio: see https://github.com/pytest-dev/pytest-asyncio#fixtures

why not use it?

Reply:
Not sure what its really give. It would just short hand in this case. If this is important in term of style I can change it. We are not looking for some optimisation here. What it dose is saving you from getting an error that no loop exists and ensure that one does.